src/Microsoft.AspNetCore.Razor.Evolution/DirectiveDescriptor.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public class DirectiveDescriptor
+    {
+        public static IDirectiveDescriptorBuilder Create(string name)
+        {
+            return new DefaultDirectiveDescriptorBuilder(name, DirectiveDescriptorType.SingleLine);
+        }
+
+        public static IDirectiveDescriptorBuilder CreateRazorBlock(string name)
+        {
+            return new DefaultDirectiveDescriptorBuilder(name, DirectiveDescriptorType.RazorBlock);
+        }
+
+        public static IDirectiveDescriptorBuilder CreateCodeBlock(string name)
+        {
+            return new DefaultDirectiveDescriptorBuilder(name, DirectiveDescriptorType.CodeBlock);
+        }
+
+        public string Name { get; set; }
+
+        public DirectiveDescriptorType Type { get; set; }
+
+        public IList<DirectiveTokenDescriptor> Tokens { get; set; }
```

Comment:
Wait, what? Is this an implementation of the builder or a factory of the builder?